tests/acceptance/bootstrap/SpacesContext.php

```diff
@@ -2065,9 +2065,9 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
 		$sourceFile = \explode("/", $sourceFile);
 		$sourceFile = \end($sourceFile);
 		$destinationFile = \trim($destinationFile, "/");
-		if ($actionType === 'copied' || $actionType === 'moved') {
+		if ($actionType === 'copied') {
 			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
-		} else {
+		} elseif ($actionType === 'renamed' || $actionType === 'moved') {
```

Comment:
was it required? and why

Reply:
Yes, because for following scenario we have to directly rename the file during the MOVE request:
``` feature
Scenario: check activities of destinaton file and folder after moving a file by renaming destination file using file-id
```
In this particular step:
``` feature 
And user "Alice" has moved file "textfile.txt" into "FOLDER/renamed.txt" inside space "Personal" using file-id "<<FILEID>>" 
```

If we didn't do it then it would return `204` instead of `201`
See: https://drone.owncloud.com/owncloud/ocis/40432/38/5

cc @saw-jan

Reply:
hmm, I think there should be another better way. I know these steps are confusing.

Reply:
Actually, what's happening is that: 
What we already have is:
```php
...
if ($actionType === 'copied' || $actionType === 'moved') {
			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
		} else {
			$fileDestination = $destinationFile;
		}
...
```
And for the steps like:
```
    And user "Alice" has copied file "textfile.txt" into "newFolder" inside space "Personal" using file-id "<<FILEID>>"
```
we are not appending the `file name` after the container i.e., `folder name` (like this: newFolder/textfile.txt). So, it gives:
```
$fileDestination: "newFolder/textfile.txt" and
$headers: Destination="https://localhost:9200/remote.php/dav/spaces/98b0f059-cf07-4bac-8528-bcaed059edcb$6ab70a1e-4a57-4542-9f76-eb9982a37da5/newFolder/textfile.txt"
```

But for scenarios like:
```
    And user "Alice" has moved file "textfile.txt" into "New Folder/textfile.txt" inside space "Personal" using file-id "<<FILEID>>"
```
we are already appending the `file name` after the container i.e., `folder name` (like this: New Folder/textfile.txt). So it gives:
```
$fileDestination = "New%20Folder/textfile.txt/textfile.txt" and
$headers: Destination="https://localhost:9200/remote.php/dav/spaces/98b0f059-cf07-4bac-8528-bcaed059edcb$6826391a-448e-42b1-9ef4-cfb39c0c4ac4/New%20Folder/textfile.txt/textfile.txt"
```
which is not expected.

So 
EITHER
 we can modify the `copy` steps appending the `file name` after container and we can totally omit above mentioned code snippet
OR
we can modify the `move` steps removing the `file name` after container and we can have the existing above mentioned code snippet
OR
we can go with changes made in this PR
OR we should dig more to find another solution

Reply:
I think, modifying the `copy` steps and omitting the snippet 
```php
if ($actionType === 'copied' || 'moved') {
			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
		} else {
			$fileDestination = $destinationFile;
		}
```
will be better choice because in other test cases also, the `file name` is appended after its container like:
https://github.com/owncloud/ocis/blob/5fc891ce94686bd89f5e6b082ca2f85c80f478f4/tests/acceptance/features/coreApiFavorites/favorites.feature#L154

Reply:
IMO, the scenario steps need to be written clearly.
Example: 
```feature
# append source <file> to the destination path (<folder>/<file>)
... (copies|moves) a file "<file>" into folder "<folder>" inside space "<space>"

# do not append source <file> to the destination path (<newfile>)
... (copies|moves|renames) a file "<file>" to "<newfile>" inside space "<space>"

# do not append source <file> to the destination path (<folder>/<newfile>)
... (copies|moves|renames) a file "<file>" to "<folder>/<newfile>" inside space "<space>"
```

Reply:
IMO, this would be better
```feature
# append source <file> to the destination path (<folder>/<file>)
... (copies|moves) a file "<file>" into folder "<folder>" inside space "<space>"

# do not append source <file> to the destination path (<newfile>)
... renames a file "<file>" to "<newfile>" inside space "<space>"

# do not append source <file> to the destination path (<folder>/<newfile>)
... renames a file "<file>" to "<folder>/<newfile>" inside space "<space>"
```

Reply:
@saw-jan Refactored the steps. Please have a look.

Reply:
@pradip did you make a separate issue / add in an existing issue for modifying steps as commented above

Reply:
> @pradip did you make a separate issue / add in an existing issue for modifying steps as commented above

I made a separate new commit for modifying the steps in this PR  and I am pushing changes that might touch previous changes in this new commit